### PR TITLE
Add network scope

### PR DESCRIPTION
**Pull request by jping0929**
Add `networkScope` to cloud query spec

### CODE REVIEW

**Comment by jping0929** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "network-scope-rql",
    "component": "backend",
    "pr-id": "1442",
    "commit-sha": "f2350edb372b7e6a1358f9209741c57ceaa125bb",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "gaia",
    "pr-id": "735",
    "commit-sha": "2787a1fe66ae2c278c5f6d22c7f6e96ad99d54a6",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "prismacloud",
    "pr-id": "68",
    "commit-sha": "d8701d1457ac0adadfafa9a209599ebc41dae426",
    "pipeline": "master"
  }
]
```

File: specs/cloudnetworkquery.spec [hunk 1]
```diff
@@ -124,6 +137,18 @@ attributes:
...
+      Indicates how source IP matching is handled. True means exact match, false means
+      subnet
+      match.
+    type: enum
+    exposed: true
+    allowed_choices:
+    - FullMatch
+    - PartialMatch
+    default_value: PartialMatch
+
```
Comment (dstiliadis) [inline specs/cloudnetworkquery.spec:151]: Unless if there is a reason that I don't understand, I think we need just one parameter networkScope and not two. Both source and destination cannot be IPs anyway and network scope only applies to IPs.

**Comment by dstiliadis** [comment]
PR also needs to add validations in the custom_validations file and associated tests. A value cannot be set if both the source and destination are selectors. It can only be set if source or destination is a selector.

**Review by dstiliadis — APPROVED** [review]

**Comment by jping0929** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "network-scope-rql",
    "component": "backend",
    "pr-id": "1442",
    "commit-sha": "92675d8822b92f3c96c5568aa02f6e9a51106122",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "gaia",
    "pr-id": "735",
    "commit-sha": "d7e7b4e45b8753d2c376b064cbb447485e9f2cdf",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "prismacloud",
    "pr-id": "68",
    "commit-sha": "6dbd10dc2e47fc3c5d21e994a6631693d005458c",
    "pipeline": "master"
  }
]
```

**Comment by jping0929** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "network-scope-rql",
    "component": "backend",
    "pr-id": "1442",
    "commit-sha": "92675d8822b92f3c96c5568aa02f6e9a51106122",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "gaia",
    "pr-id": "735",
    "commit-sha": "c6b9d26256cc4d09ed66bd10bab2c5724b6ce0bb",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "prismacloud",
    "pr-id": "68",
    "commit-sha": "d9ab77e102bbe6a006096f5fa4e2584e0e7f1d5d",
    "pipeline": "master"
  }
]
```

**Comment by jping0929** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "network-scope-rql",
    "component": "backend",
    "pr-id": "1442",
    "commit-sha": "92675d8822b92f3c96c5568aa02f6e9a51106122",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "gaia",
    "pr-id": "735",
    "commit-sha": "e3f83ca7bd7aed134225bca0af650ebf65dd8552",
    "pipeline": "master"
  },
  {
    "project": "network-scope-rql",
    "component": "prismacloud",
    "pr-id": "68",
    "commit-sha": "d9ab77e102bbe6a006096f5fa4e2584e0e7f1d5d",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]

**Pull request merged**